frontend/apps/pwa/src/components/FormPicker.tsx

```diff
@@ -1,34 +1,69 @@
-import { FC, useCallback } from 'react';
+import React, { FC, useCallback } from 'react';
 import { FormDefinition, FormType } from 'core-api-client';
+import { FieldErrors } from 'react-hook-form';
+import { ErrorMessage } from '@hookform/error-message';
 
 import { useDatabase, useForms } from '../app/hooks';
+import { ValidationForm } from '../reducers/Former/types';
+import { registeredValidation } from '../features/former/validation';
 
 export type FormPickerProps = {
   forms: FormDefinition[];
   formId: string | undefined;
   disabled?: boolean;
   setFormId: (formId: string) => void;
+  register: any;
+  errors?: FieldErrors<ValidationForm>;
 };
 
-export const FormPicker: FC<FormPickerProps> = (props) => {
-  const { forms, formId, setFormId, disabled } = props;
+export const FormPicker: FC<FormPickerProps> = ({
+  forms,
+  formId,
+  setFormId,
+  disabled,
+  register,
+  errors,
+}) => {
   const hasForms = forms.length > 0;
+
+  const registerSelectedFieldReference = register(
+    'selectedField.fieldType.reference.formId',
+    registeredValidation.selectedField.fieldType.reference.formId,
+  );
   return (
     <div>
       <select
         disabled={disabled || !hasForms}
-        onChange={(e) => setFormId(e.target.value)}
+        {...registerSelectedFieldReference}
+        onChange={(e) => {
+          setFormId(e.target.value);
+          return registerSelectedFieldReference.onChange(e);
+        }}
         value={formId || ''}
-        className="form-select"
+        className={`form-select ${
+          errors?.selectedField?.fieldType?.reference?.formId && !disabled
```

Comment:
Similarly with the errors, we should extract the specific error in the container. Again, not suggesting changes for now, just for future use.